tools/fds/server.cpp

```diff
@@ -99,15 +104,79 @@ int fastdds_discovery_server(
+            {
+                std::cout << "Cannot open XML file " << sXMLConfigFile << ". Please, check the path of this "
+                          << "XML file." << std::endl;
+                return 1;
+            }
+            if (profile.empty())
+            {
+                // Set environment variables to prevent loading the default XML file
+#ifdef _WIN32
+                if (0 != _putenv_s("FASTRTPS_DEFAULT_PROFILES_FILE", "") ||
+                        0 != _putenv_s("SKIP_DEFAULT_XML_FILE", "1"))
+                {
+                    char errmsg[1024];
+                    strerror_s(errmsg, sizeof(errmsg), errno);
+                    std::cout << "Error setting environment variables: " << errmsg << std::endl;
+                    return 1;
+                }
+#else
+                if (0 != unsetenv(fastrtps::xmlparser::DEFAULT_FASTRTPS_ENV_VARIABLE) ||
+                        0 != setenv(fastrtps::xmlparser::SKIP_DEFAULT_XML_FILE, "1", 1))
+                {
+                    std::cout << "Error setting environment variables: " << std::strerror(errno) << std::endl;
+                    return 1;
+                }
+#endif // ifdef _WIN32
+                // Set default participant QoS from XML file
+                if (ReturnCode_t::RETCODE_OK != DomainParticipantFactory::get_instance()->load_profiles())
+                {
+                    std::cout << "Error setting default DomainParticipantQos from XML default profile." << std::endl;
+                    return 1;
+                }
+                participantQos = DomainParticipantFactory::get_instance()->get_default_participant_qos();
+            }
+            else
+            {
+                DomainParticipantFactory::get_instance()->get_participant_qos_from_profile(profile, participantQos);
+            }
+        }
+
+    }
+
     // Retrieve server Id: is mandatory and only specified once
     // Note there is a specific cast to pointer if the Option is valid
     option::Option* pOp = options[SERVERID];
-    int server_id;
+    int server_id = 0;
 
     if (nullptr == pOp)
     {
-        std::cout << "Specify server id is mandatory: use -i or --server-id option." << std::endl;
-        return 1;
+        fastrtps::rtps::GuidPrefix_t prefix_cero;
+        if (!(participantQos.wire_protocol().builtin.discovery_config.discoveryProtocol ==
+                eprosima::fastrtps::rtps::DiscoveryProtocol::SERVER ||
+                participantQos.wire_protocol().builtin.discovery_config.discoveryProtocol ==
+                eprosima::fastrtps::rtps::DiscoveryProtocol::BACKUP) ||
+                participantQos.wire_protocol().prefix == prefix_cero)
+        {
+            std::cout << "The provided configuration is not valid. Participant must be either SERVER or BACKUP. " <<
+                "Server id is mandatory if not defined in the XML file: use -i or --server-id option." << std::endl;
+            return 1;
+        }
```

Comment:
I would have different checks for the discovery protocol and the guid prefix, so the error message is specific.

Reply:
Done in f4bbf6a71